pandas/core/arrays/datetimelike.py

```diff
@@ -290,14 +371,78 @@ def _box_values(self, values):
         """
         return lib.map_infer(values, self._box_func)
 
+    def _ensure_localized(self, arg, ambiguous="raise", nonexistent="raise",
+                          from_utc=False):
+        """
+        ensure that we are re-localized
+
+        This is for compat as we can then call this on all datetimelike
+        arrays generally (ignored for Period/Timedelta)
+
+        Parameters
+        ----------
+        arg : Union[DatetimeLikeArray, DatetimeIndexOpsMixin, ndarray]
+        ambiguous : str, bool, or bool-ndarray, default 'raise'
+        nonexistent : str, default 'raise'
+        from_utc : bool, default False
+            If True, localize the i8 ndarray to UTC first before converting to
+            the appropriate tz. If False, localize directly to the tz.
+
+        Returns
+        -------
+        localized DTI
+        """
+        # reconvert to local tz
+        tz = getattr(self, 'tz', None)
+        if tz is not None:
+            if not isinstance(arg, type(self)):
+                arg = self._simple_new(arg)
+            if from_utc:
+                arg = arg.tz_localize('UTC').tz_convert(tz)
+            else:
+                arg = arg.tz_localize(
+                    tz, ambiguous=ambiguous, nonexistent=nonexistent
+                )
+        return arg
+
     def __iter__(self):
         return (self._box_func(v) for v in self.asi8)
 
     @property
     def asi8(self):
+        # type: () -> ndarray
+        """
+        Integer representation of the values.
+
+        Returns
+        -------
+        ndarray
+            An ndarray with int64 dtype.
+        """
         # do not cache or you'll create a memory leak
         return self._data.view('i8')
 
+    @property
+    def _ndarray_values(self):
+        return self._data
+
+    # ------------------------------------------------------------------
+    # Formatting
+
+    def _format_native_types(self, na_rep=u'NaT', date_format=None):
+        """
+        Helper method for astype when converting to strings.
+
+        Returns
+        -------
+        ndarray[str]
+        """
+        raise AbstractMethodError(self)
+
+    def _formatter(self, boxed=False):
+        # TODO: Remove Datetime & DatetimeTZ formatters.
```

Comment:
as in remove them from pandas.io?

Reply:
Yeah, `io.formats.format` We were able to remove PeriodFormatter and InterervalFormatter. A at glance, these look a bit harder to remove (they call into cython code for some reason), so this may be tricky.